### ISSUE TITLE
Helidon is not flushing the response data and causing OutOfMemory over a period

### ISSUE THREAD

**Issue by pradeep-rout** [oracle/helidon#875]
We are observing the response data is getting accumulated in Helidon over time to cause OutOfMemory. 

## Environment Details
* Helidon Version: 0.10
* Helidon SE or Helidon MP: Helidon SE
* JDK version: 1.8.0_211
* OS: Oracle Linux
* Docker version (if applicable): 18.09.1-ol

----------
## Problem Description
Over a period of 3-4 days, one of our micro-services is kept getting OutOfMemory error. Analyzing the heap dump it looks like Helidon is not flushing the response that is already sent or for some reason. This response data is getting accumulated in byte[] and the object - io.netty.channel.socket.nio.NioSocketChannel grows more than 5.7 GB in this case. This is experienced in the container that is getting a high volume of API calls.

### Handler Implementation
```
 private void getForId(final ServerRequest request, final ServerResponse response, String table, Map<String, String> params) {
        CompletableFuture.completedFuture(table).
                thenApply(p -> {
                    GsmResponse rsp = getForId(p, params);
                    return rsp;
                }).thenCompose(p -> {
            response.status(p.getStatus()).headers().add(Http.Header.CONTENT_TYPE,
                    MediaType.APPLICATION_JSON.toString());
            return response.send(this.toJson(p)).thenAccept(g -> {
                ResponseLogging.logResponse(g);});
        }).exceptionally(t -> {
            request.next(t);
            return null;
        });
    }
```
![HeapDump-Overview](https://user-images.githubusercontent.com/49699750/62583979-81c7ad80-b867-11e9-984e-d95b12b61cb3.png)
![byte-accumulation](https://user-images.githubusercontent.com/49699750/62584000-9a37c800-b867-11e9-8129-19e565f28374.png)

**Comment by romain-grecourt** [comment]
I strongly recommend you to upgrade to the latest version of Helidon. Many performance fixes and improvements were made since version 0.1.0.

**Comment by pradeep-rout** [comment]
Thanks, Romain. We are planning to do that in a couple of weeks from now. Do you have any workaround recommendation for this issue on Helidon 0.10. Can we explicitly flush this accumulation from code as we are already in production? Any workaround would be a great help?

**Comment by spericas** [comment]
@pradeep-rout You shouldn't really go into production with Helidon **0.1.0**, as it should be clear from the version :) Helidon is backward compatible, so an upgrade should be simpler than working around this problem IMO.

**Comment by pradeep-rout** [comment]
@spericas Helidon is not backward compatible. The config API that we use has some issues in upgrading. If I’m not wrong helidon API backward compatible is from 1.x. 1.x API is not backward compatible with 0.x

**Comment by pradeep-rout** [comment]
We are using 0.10.0 not 0.1.0

**Comment by romain-grecourt** [comment]
Right, that was a typo, 0.1.0 does not exist.
You are correct, 0.x is not compatible with 1.x ; we are following semantic versioning backward is determined by the major version digit.

**Comment by spericas** [comment]
> @spericas Helidon is not backward compatible. The config API that we use has some issues in upgrading. If I’m not wrong helidon API backward compatible is from 1.x. 1.x API is not backward compatible with 0.x

 Sorry, I was just thinking mostly about the webserver API looking at your code. And, yes, I meant version 0.10.0. I'll take a closer look and see if we can backport a fix.

**Comment by spericas** [comment]
I suspect the issue is related to this PR:

https://github.com/oracle/helidon/pull/695/files

but would need some more analysis and testing to be certain.

**Comment by pradeep-rout** [comment]
Thanks @spericas

**Comment by romain-grecourt** [comment]
The reported issue is actually not about response data getting accumulated ; instead it's about accumulation of response listeners in netty when keep-alive is used. This was fixed initially in v1.1.1.

We've backported the fix to the 0.10.x code line and made a 0.10.6 patch release.
